plugins/inputs/docker/docker_test.go

```diff
@@ -1117,3 +1125,191 @@ func TestHostnameFromID(t *testing.T) {
 	}
 
 }
+
+func Test_parseContainerStatsPerDeviceAndTotal(t *testing.T) {
+	type args struct {
+		stat             *types.StatsJSON
+		acc              telegraf.Accumulator
+		tags             map[string]string
+		id               string
+		perDeviceInclude []string
+		totalInclude     []string
+		daemonOSType     string
+	}
+	stats := testStats()
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "Per device and total metrics enabled",
+			args: args{
+				stat:             stats,
+				perDeviceInclude: containerMetricClasses,
+				totalInclude:     containerMetricClasses,
+			},
+		},
+		{
+			"Per device metrics enabled",
+			args{
+				stat:             stats,
+				perDeviceInclude: containerMetricClasses,
+				totalInclude:     []string{},
+			},
+		},
+		{
+			"Total metrics enabled",
+			args{
+				stat:             stats,
+				perDeviceInclude: []string{},
+				totalInclude:     containerMetricClasses,
+			},
+		},
+		{
+			name: "Per device and total metrics disabled",
+			args: args{
+				stat:             stats,
+				perDeviceInclude: []string{},
+				totalInclude:     []string{},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var acc testutil.Accumulator
+			parseContainerStats(tt.args.stat, &acc, tt.args.tags, tt.args.id, tt.args.perDeviceInclude,
+				tt.args.totalInclude, tt.args.daemonOSType)
+
+			var numCpus int
+			if stats.CPUStats.OnlineCPUs > 0 {
+				numCpus = len(stats.CPUStats.CPUUsage.PercpuUsage[:stats.CPUStats.OnlineCPUs])
+			} else {
+				numCpus = len(stats.CPUStats.CPUUsage.PercpuUsage)
+			}
+
+			numNetworks := len(stats.Networks)
+
+			deviceStatMap := getDeviceStatMap(stats.BlkioStats)
+
+			// per device
+			for i := 0; i < numCpus; i++ {
+				if choice.Contains("cpu", tt.args.perDeviceInclude) {
+					acc.AssertTagValuesInMeasurement(t, "docker_container_cpu",
```

Comment:
I think you could achieve the same with [a cmp option](https://pkg.go.dev/github.com/google/go-cmp/cmp#FilterPath) added to `RequireMetricEqual()` instead of inventing a new function. Maybe we could add
```golang
func OnlyTags() cmp.Option {
	f := func(p cmp.Path) bool {return p.String() != "Tags"}
	return cmp.FilterPath(f, cmp.Ignore())
}
````
to testutils instead...

Reply:
Implemented this approach.

First time using options, had to add an extra condition to get it working


```
f := func(p cmp.Path) bool { return p.String() != "Tags" && p.String() != "" }
```
Does it make sense?

Reply:
Huh? If the path entry is empty we don't want to ignore it? Whatever, do whatever is necessary. I'll test the result on my side anyway... :-)

Reply:
Your code is absolutely correct. It seems that the root of a structure has an empty path. Who could have expected this. :-) Keep as is and mark as resolved.